### ISSUE TITLE
Support on Java 11

### ISSUE THREAD

**Issue by FroMage** [jklingsporn/vertx-jooq#67]
Hi,

I've tried to run the generator plugin on Java 11 and didn't find how to make it work, because Java 11 removed the `java.xml.bind` module. I'm not sure if jooq or vertx-jooq is the issue, but I wanted to ask if you support Java 11 and if yes if you have some Maven configuration you can share?

Thanks!

**Comment by jklingsporn** [comment]
Hey, thanks for your answer. The project is currently built using Java 8. Unfortunately I haven't found the time to look into the compatibility with the latest java versions

**Comment by jklingsporn** [comment]
See also: https://groups.google.com/forum/?fromgroups#!topic/jooq-user/NRtyx-hdtdA
jooq 3.12. will support java 11 until then there is not much more that I can do.

**Comment by pablotdl** [comment]
Have you tried adding the jaxb jars explicitly to your classpath? I don't know if they work in Java 11, but they are available in a maven repo.

**Comment by abonander** [comment]
@pablotdl yes that does work, if you add them to your buildscript classpath:

```groovy
buildscript {
	repositories { jcenter() }
	dependencies {
		classpath "io.github.jklingsporn:vertx-jooq-generate:4.1.0"
		classpath "org.glassfish.jaxb:jaxb-runtime:2.3.2"
	}
}
```

**Comment by aleksandar78** [comment]
Hi everyone,

a couple of months ago we have built Gradle plugin for code generation with vertx-jooq.
We had requirements to build the library with JDK 11 or above. We add a couple of dependencies for JAXB and annotation process and everything worked well. 

Today we've tried to upgrade plugin dependencies with vertx-jooq 5.0.0 version. 
We've found that JOOQ library 3.12.3 is imported as vertx-jooq transitive dependency and that generated classes have @Generated annotation with import javax.annotation as in JDK 8 instead of javax.annotation.processing.
All our environments, IDE and CI, are configured with JDK 11 and above and this kind of import brakes our project imports. 

Probably something changed with the new release but we cannot understand what.
Any suggestion welcome,

Thanks

**Comment by jklingsporn** [comment]
This might be a bug, because I remember dealing with a similar issue. The
core jooq code generator has some mechanics to detect which annotation
should be taken, but it had some issues. Will look into it.

Aleksandar <notifications@github.com> schrieb am Di., 28. Jan. 2020, 18:59:

**Comment by aleksandar78** [comment]
Thanks @jklingsporn. I found in [official JOOQ documentation ](https://www.jooq.org/doc/3.12/manual/code-generation/codegen-advanced/codegen-config-generate/codegen-generate-annotations/) a possibility to specify the version of annotation processing import.
```xml
<configuration xmlns="http://www.jooq.org/xsd/jooq-codegen-3.12.0.xsd">
  <generator>
    <generate>
      <!-- Possible values for generatedAnnotationType
         - DETECT_FROM_JDK
         - JAVAX_ANNOTATION_GENERATED
         - JAVAX_ANNOTATION_PROCESSING_GENERATED -->
      <generatedAnnotation>true</generatedAnnotation>
      <generatedAnnotationType>DETECT_FROM_JDK</generatedAnnotationType>
      
      <jpaAnnotations>true</jpaAnnotations>
      <jpaVersion>2.2</jpaVersion>
      
      <validationAnnotations>true</validationAnnotations>
      <springAnnotations>true</springAnnotations>
    </generate>
  </generator>
</configuration>
```

I'll try to set some of these values in the XML configuration file and see what happens.

**Comment by aleksandar78** [comment]
I've tried to set specific values as gneratedAnnotationType but nothing changed. Import of @Generated remains javax.annotation. 

I'll skip annotation generation for now but this remains a problem caused by some transitive dependencies.

**Comment by jklingsporn** [comment]
I've just checked this. For some reason that flag is overridden in the `VertxGenerator` (Line 603):
```
@Override
public GeneratedAnnotationType generateGeneratedAnnotationType() {
     return GeneratedAnnotationType.JAVAX_ANNOTATION_GENERATED;
}
```
I will provide a fix - hopefully this week. Until then, please try to subclass the generator of your choice, e.g. `ClassicJDBCVertxGenerator`, overwrite that method and replace the generator in your pom with the new generator.

**Comment by aleksandar78** [comment]
Thanks @jklingsporn. 
We disabled generation annotation and for now, we can generate classes and try other things.
We found another problem related to the latest vertx release with vertx-jooq library.
When we generate DAOs with our Gradle library our IDE ( Intellij Idea ) gives us the following error:
```
'queryExecutor()' in 'our.company.app.ExampleDao' clashes with 'queryExecutor()' in 'io.github.jklingsporn.vertx.jooq.shared.internal.AbstractVertxDAO'; 
attempting to use incompatible return type
```
We try to force vertx-sql client version to 3.8.5, instead of 3.8.4 that library brings as a transitive dependency, and that make generation works properly. Probably, the latest vertx versions can create some version incompatibility issues during code generation.

**Comment by jklingsporn** [comment]
I will upgrade the vertx-dependency with the next release as well and see how things are going.

**Comment by jklingsporn** [comment]
I've just pushed to https://github.com/jklingsporn/vertx-jooq/tree/release5.0.1 which contains the generated-annotation fix and bumped the vertx version to 3.8.5.
All tests ran through, so it should be fine. Can you try your project against that branch?

**Comment by aleksandar78** [comment]
Sure @jklingsporn. Tomorrow I'll pull 5.0.1 branch, build the project and deploy on our local nexus as a snapshot. I'll let you know. 
Thanks for feedback

**Comment by aleksandar78** [comment]
Hi @jklingsporn, 

We tried to compile it with JDK 11 and we found compilation errors in vertx-jooq-generate tests. All generated classes had javax.annotation.Generated import and that package missing in JDK 11. We dropped folder with test and after that build completed successfully. 

We tested with our generation Gradle plugin. With specified annotation type library generates annotation with correct import ( javax.annotation.processing ). 
Incompatibility error regarding return type disappears.

IMO version 5.0.1 works well on JDK 11.
